File: src/main/java/tn/esprit/pegasus/entities/Quiz.java
package tn.esprit.pegasus.entities;

public class Quiz {
    private int id;
    private String title;
    private Integer timeLimitMin;
    private int passingScore;
    private Integer attemptLimit;
    private int courseId;

    public Quiz() {
    }

    public Quiz(int id, String title, Integer timeLimitMin, int passingScore, Integer attemptLimit, int courseId) {
        this.id = id;
        this.title = title;
        this.timeLimitMin = timeLimitMin;
        this.passingScore = passingScore;
        this.attemptLimit = attemptLimit;
        this.courseId = courseId;
    }

    public Quiz(String title, Integer timeLimitMin, int passingScore, Integer attemptLimit, int courseId) {
        this.title = title;
        this.timeLimitMin = timeLimitMin;
        this.passingScore = passingScore;
        this.attemptLimit = attemptLimit;
        this.courseId = courseId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Integer getTimeLimitMin() {
        return timeLimitMin;
    }

    public void setTimeLimitMin(Integer timeLimitMin) {
        this.timeLimitMin = timeLimitMin;
    }

    public int getPassingScore() {
        return passingScore;
    }

    public void setPassingScore(int passingScore) {
        this.passingScore = passingScore;
    }

    public Integer getAttemptLimit() {
        return attemptLimit;
    }

    public void setAttemptLimit(Integer attemptLimit) {
        this.attemptLimit = attemptLimit;
    }

    public int getCourseId() {
        return courseId;
    }

    public void setCourseId(int courseId) {
        this.courseId = courseId;
    }

    @Override
    public String toString() {
        return "Quiz{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", passingScore=" + passingScore +
                ", courseId=" + courseId +
                '}';
    }
}

File: WEB_Application/src/Entity/Quiz.php
<?php

namespace App\Entity;

use App\Repository\QuizRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: QuizRepository::class)]
class Quiz
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank]
    private ?string $title = null;

    #[ORM\Column(nullable: true)]
    private ?int $timeLimitMin = null;

    #[ORM\Column]
    #[Assert\Range(min: 0, max: 100)]
    private ?int $passingScore = null;

    #[ORM\Column(nullable: true)]
    private ?int $attemptLimit = null;

    #[ORM\ManyToOne(inversedBy: 'quizzes')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Course $course = null;

    /**
     * @var Collection<int, QuizQuestion>
     */
    #[ORM\OneToMany(targetEntity: QuizQuestion::class, mappedBy: 'quiz', orphanRemoval: true)]
    private Collection $quizQuestions;

    public function __construct()
    {
        $this->quizQuestions = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(string $title): static
    {
        $this->title = $title;

        return $this;
    }

    public function getTimeLimitMin(): ?int
    {
        return $this->timeLimitMin;
    }

    public function setTimeLimitMin(?int $timeLimitMin): static
    {
        $this->timeLimitMin = $timeLimitMin;

        return $this;
    }

    public function getPassingScore(): ?int
    {
        return $this->passingScore;
    }

    public function setPassingScore(int $passingScore): static
    {
        $this->passingScore = $passingScore;

        return $this;
    }

    public function getAttemptLimit(): ?int
    {
        return $this->attemptLimit;
    }

    public function setAttemptLimit(?int $attemptLimit): static
    {
        $this->attemptLimit = $attemptLimit;

        return $this;
    }

    public function getCourse(): ?Course
    {
        return $this->course;
    }

    public function setCourse(?Course $course): static
    {
        $this->course = $course;

        return $this;
    }

    /**
     * @return Collection<int, QuizQuestion>
     */
    public function getQuizQuestions(): Collection
    {
        return $this->quizQuestions;
    }

    public function addQuizQuestion(QuizQuestion $quizQuestion): static
    {
        if (!$this->quizQuestions->contains($quizQuestion)) {
            $this->quizQuestions->add($quizQuestion);
            $quizQuestion->setQuiz($this);
        }

        return $this;
    }

    public function removeQuizQuestion(QuizQuestion $quizQuestion): static
    {
        if ($this->quizQuestions->removeElement($quizQuestion)) {
            // set the owning side to null (unless already changed)
            if ($quizQuestion->getQuiz() === $this) {
                $quizQuestion->setQuiz(null);
            }
        }

        return $this;
    }
}
